crates/matrix-sdk-crypto/src/session_manager/sessions.rs

```diff
@@ -347,57 +355,13 @@ impl SessionManager {
             }
         };
 
-        // Collect the (user_id, device_id, device_key_id) triple for logging reasons.
-        let one_time_keys: BTreeMap<_, BTreeMap<_, BTreeSet<_>>> = response
-            .one_time_keys
-            .iter()
-            .map(|(user_id, device_map)| {
-                (
-                    user_id,
-                    device_map
-                        .iter()
-                        .map(|(device_id, key_map)| {
-                            (device_id, key_map.keys().collect::<BTreeSet<_>>())
-                        })
-                        .collect::<BTreeMap<_, _>>(),
-                )
-            })
-            .collect();
-
-        debug!(?request_id, ?one_time_keys, failures = ?response.failures, "Received a `/keys/claim` response");
-
-        let failed_servers: BTreeSet<_> = response
-            .failures
-            .keys()
-            .filter_map(|s| ServerName::parse(s).ok())
-            .filter(|s| s != self.store.static_account().user_id.server_name())
-            .collect();
-        let successful_servers = response.one_time_keys.keys().map(|u| u.server_name());
-
-        self.failures.extend(failed_servers.iter().cloned());
-        self.failures.remove(successful_servers);
-
-        // build a map of (user_id, device_id) -> key from the response
-        let sessions_to_create: BTreeMap<(_, _), _> = response
-            .one_time_keys
-            .iter()
-            .flat_map(|(user_id, device_map)| {
-                device_map.iter().filter_map(|(device_id, key_map)| {
-                    // For devices that returned at least one key, this will yield
-                    // `Some((user_id, device_id), first_key)`. For devices that returned an
-                    // empty map, this will yield `None`.
-                    //
-                    // The `None` entries are then filtered out by `filter_map`.
-                    key_map.values().next().map(|key| ((user_id.clone(), device_id.clone()), key))
-                })
-            })
-            .collect();
-
-        // if we were able to pair this response with a request, look for devices that
+        // If we were able to pair this response with a request, look for devices that
         // were present in the request but did not elicit a successful response.
         if let Some(request) = request {
-            let devices_in_response: BTreeSet<_> =
-                sessions_to_create.iter().map(|((u, d), _)| (u, d)).collect();
+            let devices_in_response: BTreeSet<_> = one_time_keys
+                .iter()
+                .flat_map(|(u, d)| d.keys().map(|d| (*u, *d)).collect::<BTreeSet<_>>())
```

Comment:
```suggestion
                .flat_map(|&(u, d)| d.keys().map(|d| (u, d)).collect::<BTreeSet<_>>())
```

Also the one-character variable names are not really clear, especially as `d` is reused.

Reply:
That suggestion doesn't seem to work. I'll change the names though.